apps/chat/src/components/Chat/ChatHeader/Header.tsx

```diff
@@ -170,9 +170,9 @@ export const ChatHeader = Inversify.register(
       screenState === ScreenState.SM && conversationSelectedAddons.length > 2;
     const isConversationInvalid = isEntityNameOrPathInvalid(conversation);
 
-    const disallowChangeAgent = isChangeAgentDisallowed || isExternal;
+    const disallowChangeAgent = isChangeAgentDisallowed || isReadOnly;
```

Comment:
Looks like it should not be allowed to change agent for external conversations, because it can stop sharing